app/lib/discord/events/puzzle_answer.rb

```diff
@@ -40,23 +40,29 @@ def handle(event)
           return
         end
 
-        server = Server.find_or_initialize_by(server_id: event.server.id)
-
         # Create the answer record
-        answer = Answer.create!(
+        answer = Answer.create(
           puzzle_id: puzzle.id,
           user_id: user.id,
           server_id: server.id,
           choice: answer, # 'ruby' or 'rails'
           is_correct: puzzle.answer.to_s == answer # Correct answer check
         )
-
-        # Respond to the user with the result
-        event.respond(
-          content: nil,
-          embeds: [ answer_embed(answer, puzzle) ],
-          ephemeral: true # Only the user sees this message
-        )
+        if answer
+          # Respond to the user with the result
+          event.respond(
+            content: nil,
+            embeds: [ answer_embed(answer, puzzle) ],
+            ephemeral: true # Only the user sees this message
+          )
+        else
```

Comment:
After looking further into this it seems as thouse answer should always be nil if the validation fails or the db index fails, so the if/else should work here.